### ISSUE TITLE
Release of 0.23 broke 0.22 build

### ISSUE THREAD

**Issue by Florob** [libpnet/libpnet#391]
The release of 0.23 seems to have broken 0.22 builds.
The specific error is
```
error[E0599]: no method named `to_primitive_values` found for type `pnet_base::MacAddr` in the current scope
   --> …/target/armv7-unknown-linux-gnueabihf/release/build/pnet_packet-242d82d1735b29fa/out/ethernet.rs:442:24
    |
442 |         let vals = val.to_primitive_values();
    |                        ^^^^^^^^^^^^^^^^^^^ method not found in `pnet_base::MacAddr`
```

After chasing the dependency chain my understanding of this is as follows:
pnet 0.22 depends on pnet_packet 0.22. This in turn depends on pnet_macros_support 0.22 and pnet_base 0.22.
pnet_macros_support however depends on pnet_base >= 0.22.
With the release of 0.23 pnet_macros_support therefore began pulling in pnet_base 0.23. This means that the `ToPrimitives` implementation in pnet_macros_support is now for the types defined in pnet_base 0.23, not the ones in pnet_base 0.22.

A fix would likely be releasing pnet_macro_support 0.22.1 that depends on pnet_base 0.22 instead of pnet_base >= 0.22.

**Comment by JuxhinDB** [comment]
@mrmonday Would you like me to handle this in a similar fashion?

**Comment by ctacke** [comment]
In the interim, you can get past the error by doing this pre-build:

`cargo update -p pnet_base:0.23.0 --precise 0.22.0`

**Comment by imsnif** [comment]
Just a heads up, the above solution doesn't work for binary packages without a build step (since their Cargo.lock file is generated by crates.io). I had to solve this by upgrading to 0.23.0.

**Comment by JuxhinDB** [comment]
This should now be fixed with 0.25.0. Sorry for the slow reaction to this. Hope it helps :-)
